src/java.base/windows/classes/java/io/WinNTFileSystem.java

```diff
@@ -365,24 +365,31 @@ public boolean isInvalid(File f) {
         if (f.getPath().indexOf('\u0000') >= 0)
             return true;
 
+        // Invalid if the pathname or a directory in it has a trailing space
+        File theFile = f;
+        do {
+            String p = theFile.getPath();
+            int len = p.length();
+            if (len > 0 && p.charAt(len - 1) == ' ')
+                return true;
+            theFile = theFile.getParentFile();
+        } while (theFile != null);
```

Comment:
The input has already been normalized so I think you can reduce it down to searching getPath for a trailing space or a backslash followed by a space.

Reply:
Using `Path.of` with various inputs yields the following:
```
"root\dir\subdir\file.txt" is valid
"root \dir\subdir\file.txt" is invalid
"root\ dir\subdir\file.txt" is valid
"root\dir \subdir\file.txt" is invalid
"root\dir\ subdir\file.txt" is valid
"root\dir\subdir \file.txt" is invalid
"root\dir\subdir\ file.txt" is valid
"root\dir\subdir\file.txt " is invalid
```
Apparently that method at least thinks that a backslash followed by a space is acceptable ("invalid" == `InvalidPathException` caught).